File: 1 Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> h;
        int comp;
        for (int i=0; i<nums.size(); i++){
            comp = target - nums[i];
            if (h.count(comp)){
                return vector<int> {h[comp], i};
            }
            h.insert({nums[i], i});
        }
        return vector<int> {0, 0};
    }
};

File: 1 Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, n in enumerate(nums):
            if n in d:
                return [i, d[n]]
            else:
                d[target - n] = i 
        return [0, 0]